src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1914,13 +1917,15 @@ private static Binder GetLambdaEnclosingBinder(int position, CSharpSyntaxNode st
         /// </summary>
         protected void EnsureNullabilityAnalysisPerformedIfNecessary()
         {
-            // If we're in DEBUG mode, always enable the analysis, but throw away the results
-#if !DEBUG
-            if (!Compilation.NullableSemanticAnalysisEnabled)
+            // If we're in DEBUG mode, enable the analysis unless explicitly disabled, but throw away the results
+            if (!Compilation.IsNullableAnalysisEnabled
+#if DEBUG
+                && Compilation.IsNullableAnalysisExplicitlyDisabled
```

Comment:
Shouldn't this be `||`?

Reply:
We only want to skip analysis in debug builds if `"run-nullable-analysis=false"`. In that case, `IsNullableAnalysisEnabled` returns `false` and `IsNullableAnalysisExplicitlyDisabled` returns `true`. The first check is actually redundant in this case.

---
In reply to: [541209579](https://github.com/dotnet/roslyn/pull/49876#discussion_r541209579) [](ancestors = 541209579)

Reply:
I find the `#if` in the middle of an `if` condition really hard to grok. Can we break this up so it's not splitting conditions?